bin/pycbc_inspiral

```diff
@@ -371,106 +444,48 @@ with ctx:
-                         (t_num + 1, len(bank), s_num + 1, len(segments)))
-
-            nfilters = nfilters + 1
-            snr, norm, corr, idx, snrv = \
-               matched_filter.matched_filter_and_cluster(s_num,
-                                                         template.sigmasq(stilde.psd),
-                                                         cluster_window,
-                                                         epoch=stilde._epoch)
-
-            if not len(idx):
-                continue
-
-            out_vals['bank_chisq'], out_vals['bank_chisq_dof'] = \
-                  bank_chisq.values(template, stilde.psd, stilde, snrv, norm,
-                                    idx+stilde.analyze.start)
-
-            out_vals['chisq'], out_vals['chisq_dof'] = \
-                  power_chisq.values(corr, snrv, norm, stilde.psd,
-                                     idx+stilde.analyze.start, template)
-
-            out_vals['sg_chisq'] = sg_chisq.values(stilde, template, stilde.psd,
-                                          snrv, norm,
-                                          out_vals['chisq'],
-                                          out_vals['chisq_dof'],
-                                          idx+stilde.analyze.start)
-
-            out_vals['cont_chisq'] = \
-                  autochisq.values(snr, idx+stilde.analyze.start, template,
-                                   stilde.psd, norm, stilde=stilde,
-                                   low_frequency_cutoff=flow)
-
-            idx += stilde.cumulative_index
-
-            out_vals['time_index'] = idx
-            out_vals['snr'] = snrv * norm
-
-            if opt.psdvar_short_segment is not None:
-                out_vals['psd_var_val'] = \
-                            pycbc.psd.find_trigger_value(psd_var,
-                                          out_vals['time_index'],
-                                          opt.gps_start_time, opt.sample_rate)
-
-            event_mgr.add_template_events(names, [out_vals[n] for n in names])
-
-        event_mgr.cluster_template_events("time_index", "snr", cluster_window)
-        event_mgr.finalize_template_events()
+    
+    tanalyze = list(range(tnum_start, len(bank)))
+    n = opt.template_groupby
+    tchunks = [tanalyze[i:i + n] for i in range(0, len(tanalyze), n)] 
+    
+    mmap = map
+    if opt.multiprocessing:
+        mmap = Pool(opt.multiprocessing).map
+    
+    for tchunk in tchunks:
+        data = list(mmap(template_triggers, tchunk))
+
+        for elem in data:
+            out_vals_all, tparam = elem
+            if len(out_vals_all) > 0:
+                event_mgr.new_template(tmplt=tparam)
+
+                for edata in out_vals_all:
+                    event_mgr.add_template_events(names, [edata[n] for n in names])              
+                                    
+                event_mgr.cluster_template_events("time_index", "snr", cluster_window)
```

Comment:
This is a bit hard to parse in the diff. But is it the case that `consolidate_events` will run *outside* the for loop `for elem in data` but *within* the `for tchunk in tchunks` for loop? I'm not sure `(tchunk[0]) % opt.finalize_events_template_rate` is then the right check for running consolidate events as I think there are cases where that never parses to True?

It feels like  `opt.template_groupby` and `opt.finalize_events_template_rate` might actually be combined into the one option? "Run this many templates and then consolidate" (either by stopping the multi-processing pool, and/or just by clustering triggers). I can't really see a use case where this might be set to *different* values?

Reply:
... Although perhaps having `finalize_events_template_rate` run for every N *groups* of triggers, might also make sense. Checkpointing perhaps needs to occur less often than clustering?

Reply:
@spxiwh Yes, that was one of the issue I was pointing out before that if you don't set the options very specifically, I think consolidate might not run.

@spxiwh Either option you propose (i.e. combining, or having finalize act on 'groups' rather than individual templates) would be fine with me. Combining is simpler, but does it server your use case (I think @josh-willis added this function). Combining does had some restriction on the range of 'good' values, but my guess is they were in the same range already. 

 Checkpointing doesn't happen at the same rate anyway, it's just the check if it needs to do a checkpoint which does. So as long as it is not so long that it will never check, that's fine from the checkpoint perspective.

Reply:
@ahnitz I think I was the one who moved `consolidate_events` into the loop over templates, and this was done to control memory usage in particularly noisy stretches of data (it's pointless storing in memory triggers that will be clustered away later). This feels to me like it's tied to the same decision as "how many templates should I run in a parallel (or not) pool before consolidating", so I'd be happy to combine these.

Reply:
@spxiwh Ok, great. I'll combine those options then.

Reply:
Ok, I've updated to just use the --finalize option as this value.